### ISSUE TITLE
SIP/2.0 401 Unauthorized

### ISSUE THREAD

**Issue by kiritnim** [datso/react-native-pjsip#182]
Hi,

I have the error to create an account.

> pjsua_core.c  .RX 499 bytes Response msg 401/REGISTER/cseq=39923 (rdata0x11c08e228) from UDP 131.191.161.131:5060:
> SIP/2.0 401 Unauthorized

**My Account configuration :** 
```

let configuration = {
      "name": "John1",
      "username": "0662505070",
      "domain": "131.191.161.131",
      "password": "******",
      "proxy": null,
      "transport": "UDP", // Default TCP
      "regServer": null, // Default wildcard
      "regTimeout": 3600 // Default 3600
    };

```

Please help me.

Thanks.

**Comment by yunhanxia** [comment]
I think you forgot to add the port, your domain should be look like this "131.191.161.131:5161"

**Comment by kiritnim** [comment]
> I think you forgot to add the port, your domain should be look like this "131.191.161.131:5161"

I have added port like you say, I got message "Invalid Domain". I think domain is right because account is successfully created use of the domain.

**Comment by yunhanxia** [comment]
about your username, i don't think you can use a number from trunk, try something local, like "something#123", also make sure you configured for PJSIP

**Comment by kiritnim** [comment]
> about your username, i don't think you can use a number from trunk, try something local, like "something#123", also make sure you configured for PJSIP

Can you send me full configuration you set for PJSIP.

**Comment by yunhanxia** [comment]
let configuration = {
                "name": "something#extension",
                "username": "something#extension",
                "transport": "UDP",
                "domain": "address:port",
                "password": "password"
            };
Did you configure PJSIP at asterisk side ?

**Comment by kiritnim** [comment]
> let configuration = {
> "name": "something#extension",
> "username": "something#extension",
> "transport": "UDP",
> "domain": "address:port",
> "password": "password"
> };
> Did you configure PJSIP at asterisk side ?

I thinks username is writ

**Comment by kiritnim** [comment]
@yunhanxia  PJSIP at asterisk side ? I don't know.

**Comment by yunhanxia** [comment]
your server I mean

**Comment by kiritnim** [comment]
okay. Can you send me server setting for it.

**Comment by yunhanxia** [comment]
Can you send the full code of creating account and make call ?

**Comment by kiritnim** [comment]
> Can you send the full code of creating account and make call ?

I send some code.

**Comment by kiritnim** [comment]
```
import { Endpoint } from 'react-native-pjsip'
let endpoint = new Endpoint();

    let configuration = {
      "name": "name",
      "username": "username",
      "domain": "domain",
      "password": "password",
      "proxy": null,
      "transport": "UDP", // Default TCP
      "regServer": null, // Default wildcard
      "regTimeout": 3600 // Default 3600
    };

    endpoint.createAccount(configuration).then((account) => {
      console.log("Account created", account);
      if (account !== null) {
                let call = await endpoint.makeCall(this.state.account, +9157665544);
      }
    });
```

As Above code for creating Account and for make a call after successfully creating account.

**Comment by yunhanxia** [comment]
let options = {
            headers: {
                'P-Asserted-Identity': 'Header Example',
                'X-UA': 'React native'
            }
        };
        let call = await endpoint.makeCall(account, "sip:" + number + "@" + domain, options);
thats my makecall

**Comment by kiritnim** [comment]
The Header value is fixed. 
```
'P-Asserted-Identity': 'Header Example',
'X-UA': 'React native'

```

**Comment by kiritnim** [comment]
I check on LinePhone App with all setting. In this app account created successfully but there is not calling inside the app. 

configure App : https://play.google.com/store/apps/details?id=org.linphone&hl=en_IN

Can you say what's issue. and what's server side integration.

**Comment by yunhanxia** [comment]
I don't have access to my server's source config (asterisk), but we have an admin platform that i can update an extension on PJSIP or SIP.
In my App, I did a local config with username as "something#extension" (myhome#500 for example) and domain as "myipaddress:port" (192.168.1.25:5161 for example).

For make call, my destination : "sip:" + number + "@" + domain", the number could be local and external
(sip:600@192.168.1.25:5161 or sip:0612345678@192.168.1.25:5161)

**Comment by itolegmysin** [comment]
Hey, I've the same problem.
I started the officeSIP server (windows 10) and am trying to connect to an existing user and pass following configuration:
```
const configuration = {
      name: "Oleg",
      username: "oleg",
      domain: '192.168.4.179:5060',
      password: "123",
      proxy: null,
    };
```

![image](https://user-images.githubusercontent.com/25920773/68208014-53ca4480-ffe1-11e9-9f33-e56fc50bec1c.png)

Versions:
react-native: **0.59.9**
react-native-pjsip: **2.7.4**

**Comment by Amaan630** [comment]
How did you manage to successfully install this? When I run `npm i react-native-pjsip` I am getting version ^2.7.4 but there is a podspec issue that prevents me from using it. 2.8.0 appears to have resolved this issue but npm is not installing 2.8.0.

**Comment by marufMunshi** [comment]
@Amaan630  
`"react-native-pjsip": "git+https://github.com/datso/react-native-pjsip.git"`
this will install 2.8.0

**Comment by marufMunshi** [comment]
Hi, @yunhanxia 
I am facing this issue 

`Title:       pjsua_call_make_call(acc.getId(), &pj_dst_uri, param.p_opt, this, param.p_msg_data, &id)
Code:        171039
Description: Invalid URI (PJSIP_EINVALIDURI)
Location:    ../src/pjsua2/call.cpp:566`

Thanks in advance.

**Comment by Sulbha-Mishra29** [comment]
Hi @marufMunshi 
I am facing the same issue, have you find out the solution ??
Thanks in advance.

**Comment by marufMunshi** [comment]
No, I did not find the solution, we eventually ended up using Linphone. 
@Sulbha-Mishra29

**Comment by Sulbha-Mishra29** [comment]
Okay thanks for reply, But Linphone is only available for Android project,
So have you created interface for react native application ??

**Comment by marufMunshi** [comment]
Yes, but not by me. I don't have any experience with Android. 
This is done by another developer in my company.
@Sulbha-Mishra29
